Sdk basefiles

karthikamurthy opened the pull request:
Added interface, jrpc folder from Github Avalon repo.
Repo: https://github.com/hyperledger/avalon.git
Folder path: 
1) avalon/sdk/avalon_sdk/connector/interfaces
2) avalon/sdk/avalon_sdk/connector/direct/jrpc
Modified: No

Ram-srini committed:
Changes to refactor connector
1. Create connector module in sdk and move direct model implementation classes to connector.
2. Create blockchains module in connector and move fabric, ethereum implementation classes to blockchains.
3. Fix imports

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

Ram-srini committed:
Changes to refactor connector
1. Create connector module in sdk and move direct model implementation classes to connector.
2. Create blockchains module in connector and move fabric, ethereum implementation classes to blockchains.
3. Fix imports

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

rranjan3 committed:
Move HTTPClient to common python

- Move http client to common/python from sdk
- Refator existing code to update this
- Remove HTTPClient from ethereum listener. Dead code.

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

rranjan3 committed:
Move HTTPClient to common python

- Move http client to common/python from sdk
- Refator existing code to update this
- Remove HTTPClient from ethereum listener. Dead code.

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

rranjan3 committed:
Add retry to http client

 - Add retry mechanism to HttpJrpcClient
 - Default to no retries
 - Pass config in WPE to retry in case of failures to connect
 - Remove temporary sleep in WPE start in docker-compose-pool

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

Ram-srini committed:
Remove json serialization in jrpc worker registry

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

danintel reviewed (changes requested):
I see lookup tag spelled as `lookUpTag` or `lookupTag`. You should be consistent in the code (and also the comments) as this can be the cause of errors and bugs.

karthikamurthy committed:
Added interface, jrpc folder from Github Avalon repo.
Repo: https://github.com/hyperledger/avalon.git
folder path:
1) avalon/sdk/avalon_sdk/connector/interfaces
2) avalon/sdk/avalon_sdk/connector/direct/jrpc
modified: No

Signed-off-by: Karthika Murthy <karthika.murthy@intel.com>

karthikamurthy commented:
Done

karthikamurthy commented:
> I see lookup tag spelled as `lookUpTag` or `lookupTag`. You should be consistent in the code (and also the comments) as this can be the cause of errors and bugs.

Done
